BrowserKit/Sources/SiteImageView/ImageProcessing/SiteImageFetcher/SiteImageDownloader.swift

```diff
@@ -11,72 +11,47 @@ import Common
 /// Image downloader wrapper around Kingfisher image downloader
 /// Used in FaviconFetcher
 protocol SiteImageDownloader: AnyObject {
-    /// Provides the KingFisher ImageDownloader with a Timeout in case the completion isn't called
-    var timeoutDelay: UInt64 { get }
-    var continuation: CheckedContinuation<any SiteImageLoadingResult, any Error>? { get set }
+    /// Specifies the timeout on image downloads
+    var timeoutDelay: Double { get }
     var logger: Logger { get }
 
-    @discardableResult
-    func downloadImage(
-        with url: URL,
-        completionHandler: ((Result<SiteImageLoadingResult, Error>) -> Void)?
-    ) -> DownloadTask?
-
     func downloadImage(with url: URL) async throws -> SiteImageLoadingResult
 }
 
 extension SiteImageDownloader {
+    /// Downloads an image at the given URL. Throws `SiteImageError` errors.
     func downloadImage(with url: URL) async throws -> SiteImageLoadingResult {
-        return try await withThrowingTaskGroup(of: SiteImageLoadingResult.self) { group in
-            // Use task groups to have a timeout when downloading an image from Kingfisher
-            // due to https://sentry.io/share/issue/951b878416374dd98eccb6fd88fd8427
-            group.addTask {
-                return try await self.handleImageDownload(url: url)
-            }
-
-            group.addTask {
-                try await self.handleTimeout()
-            }
-
-            // wait for the first task and cancel the other one
-            let result = try await group.next()
-            group.cancelAll()
-            guard let result = result else {
-                throw SiteImageError.unableToDownloadImage("Result not present")
-            }
-            return result
+        // Override Kingfisher's default timeout (which is 15s)
+        let modifier = AnyModifier { request in
+            var r = request
```

Comment:
nit: could we make these more verbose please (instead of one letter)

Reply:
Oops yep, I'll push that up, that was a copy & paste from the Kingfisher docs.